executor/show_stats.go

```diff
@@ -0,0 +1,178 @@
+				}
+				for _, idx := range statsTbl.Indices {
+					e.rows = append(e.rows, e.histogramToRow(db.Name.O, tbl.Name.O, idx.Info.Name.O, 1, idx.Histogram))
+				}
+			}
+		}
+	}
+	return nil
+}
+
+func (e *ShowExec) histogramToRow(dbName string, tblName string, colName string, isIndex int, hist statistics.Histogram) *Row {
+	return &Row{
+		Data: types.MakeDatums(
+			dbName,
+			tblName,
+			colName,
+			isIndex,
+			e.versionToTime(hist.LastUpdateVersion),
+			hist.NDV,
+			hist.NullCount,
+		),
+	}
+}
+
+func (e *ShowExec) versionToTime(version uint64) types.Time {
+	t := time.Unix(0, oracle.ExtractPhysical(version)*int64(time.Millisecond))
+	return types.Time{Time: types.FromGoTime(t), Type: mysql.TypeDatetime}
+}
+
+func (e *ShowExec) fetchShowStatsBuckets() error {
+	do := sessionctx.GetDomain(e.ctx)
+	h := do.StatsHandle()
+	dbs := do.InfoSchema().AllSchemas()
+	for _, db := range dbs {
+		for _, tbl := range db.Tables {
+			statsTbl := h.GetTableStats(tbl.ID)
+			if !statsTbl.Pseudo {
+				for _, col := range statsTbl.Columns {
+					rows, err := e.bucketsToRows(db.Name.O, tbl.Name.O, col.Info.Name.O, 0, col.Histogram)
+					if err != nil {
+						return errors.Trace(err)
+					}
+					e.rows = append(e.rows, rows...)
+				}
+				for _, idx := range statsTbl.Indices {
+					rows, err := e.bucketsToRows(db.Name.O, tbl.Name.O, idx.Info.Name.O, len(idx.Info.Columns), idx.Histogram)
+					if err != nil {
+						return errors.Trace(err)
+					}
+					e.rows = append(e.rows, rows...)
+				}
+			}
+		}
+	}
+	return nil
+}
+
+// bucketsToRows converts histogram buckets to rows. If the histogram is built from index, then numOfCols equals to number
+// of index columns, else numOfCols is 0.
+func (e *ShowExec) bucketsToRows(dbName, tblName, colName string, numOfCols int, hist statistics.Histogram) ([]*Row, error) {
+	isIndex := 0
+	if numOfCols > 0 {
+		isIndex = 1
+	}
+	var rows []*Row
+	for i, bkt := range hist.Buckets {
+		lowerBoundStr, err := e.valueToString(bkt.LowerBound, numOfCols)
+		if err != nil {
+			return nil, errors.Trace(err)
+		}
+		upperBoundStr, err := e.valueToString(bkt.UpperBound, numOfCols)
+		if err != nil {
+			return nil, errors.Trace(err)
+		}
+		row := &Row{
+			Data: types.MakeDatums(
+				dbName,
+				tblName,
+				colName,
+				isIndex,
+				i,
+				bkt.Count,
+				bkt.Repeats,
+				lowerBoundStr,
+				upperBoundStr,
+			),
+		}
+		rows = append(rows, row)
+	}
+	return rows, nil
+}
+
+// valueToString converts a possible encoded value to a formatted string. If the value is encoded, then
+// size equals to number of origin values, else size is 0.
+func (e *ShowExec) valueToString(value types.Datum, size int) (string, error) {
+	if size == 0 {
+		return value.ToString()
+	}
+	decodedVals, err := codec.Decode(value.GetBytes(), size)
+	if err != nil {
+		return "", errors.Trace(err)
+	}
+	var strs []string
+	for _, val := range decodedVals {
```

Comment:
I think this can be extracted as function `DatumsToString` in package `types.datum` or there already is one?